src/NuGet.Core/NuGet.Build.Tasks/GetRestoreSettingsTask.cs

```diff
@@ -108,18 +109,21 @@ public override bool Execute()
                 }
 
                 // Settings
-                var settings = RestoreSettingsUtils.ReadSettings(RestoreSolutionDirectory, Path.GetDirectoryName(ProjectUniqueName), RestoreConfigFile, _machineWideSettings);
+                // Find the absolute path of nuget.config, this should only be set on the command line. Setting the path in project files
+                // is something that could happen, but it is not supported.
+                var absoluteConfigFilePath = GetGlobalAbsolutePath(RestoreConfigFile);
+                var settings = RestoreSettingsUtils.ReadSettings(RestoreSolutionDirectory, Path.GetDirectoryName(ProjectUniqueName), absoluteConfigFilePath, _machineWideSettings);
```

Comment:
Since both RestoreSolutionDirectory and ConfigFile absolute path are set...the restoresettingsutils will create a settings object that contains both the potential solution config file and the passed config file. 

Is that behavior consistent with 4.1.0?
Should we reconsider it?

Reply:
Cleared up offline...there's a case where basically we say use the solutiondir only if the config file is not set.